### PR TITLE
minor command line improvements

### PR DESCRIPTION
**Pull request by mitchcapper**
specifying multiple hostsnames on the command line (set the cn to the first and the hosts to all for dns), specifying a custom crl url on the command line

### CODE REVIEW

File: signer/signer.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type SignRequest struct {
 	Request    string      `json:"certificate_request"`
 	Subject    *Subject    `json:"subject,omitempty"`
 	Profile    string      `json:"profile"`
+	CRLOverride	 string	`json:"crl_override"`
```
Comment (grittygrease) [inline signer/signer.go]: Run gofmt on this file, please.

**Comment by mitchcapper** [comment]
all set ran gofmt as requested.

**Comment by kisom** [comment]
Setting the CN to the first hostname isn't something we'd like in CFSSL.

The [current CAB Forum baseline requirements](https://cabforum.org/wp-content/uploads/Baseline_Requirements_V1_3_1.pdf) (if you're particularly curious, section 7.1.4.2.2a) note that the common name field is deprecated; while its use is not prohibited, it's discouraged. On a similar note, [RFC 6125](https://tools.ietf.org/html/rfc6125#section-6.4.4) prohibits (MUST NOT) the use of the field for matching a domain name when a DNS SAN is present. As the SAN is required for a BR-compliant certificate, there's no reason to use the commonName field, and we'd prefer to follow the CAB Forum's guidelines in not using the field. For certificates that aren't meant to secure traffic for a browser (such as a client authentication certificate), the signer will still copy over the common name from the CSR.

**Comment by mitchcapper** [comment]
OK, but currently it just sets it to the entire comma separated domain list.   Honestly I am just trying to largely fix the domain bug here.  Documentation officially says comma separate it to do multiple domains ("[-hostname comma,separated,hostnames]") but that doesn't currently work.   Are you asking I modify it to blank the CN field on the certificates?

**Comment by kisom** [comment]
We'd rather just not touch the CN there.

**Comment by grittygrease** [comment]
@mitchcapper exactly. Please fix the comma separated list bug and leave the CN parsing as it currently is. Thanks for your PR!

**Comment by mitchcapper** [comment]
Sorry for the delay, we implemented this awhile ago and I was trying to figure out why we had that line, at first I figured the CN was picking up the entire list.   It turns out we had a desire to set the CN on the command line outside of the json file and there is no command to do so.  

If this would be useful I am happy to submit a PR, with the desired implementation.  I could see either a separate flag specifying the CN or just a flag to use the hostname for the CN (and then I would assume the same splitting would be desired).  If this is not desired then we will just use a private patch for our version.

**Comment by grittygrease** [comment]
I think a CN flag to fix your issue would work. Happy to review a PR for just that.

**Comment by mitchcapper** [comment]
I rebased this onto HEAD and had previously removed the adjusting of the CN field.  I also submitted PR #591 which adds the ability to override the CN.  Anything else needed for this PR?

**Comment by kisom** [comment]
The tests are failing because `gofmt` needed to fix a file. I did some digging around, and it's line 101 of local.go --- if you could run `go fmt` (or better, `goimports -l -w`) on signer/local/local.go, the tests should pass (I don't have push access to your branch, so I can't make the change myself).

**Comment by mitchcapper** [comment]
the space bit me. go fmt run and pushed.

**Comment by kisom** [comment]
LGTM.

**Pull request merged**